### ISSUE TITLE
Rule Proposal: `no-extra-label`

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#5059]
From #5052.

This rule warns labels of `break` and `continue` if we can remove the label merely.

The following patterns are considered problems:

```js
/*eslint no-extra-label: 2*/

A:
for (let i = 0; i < 10; ++i) {
    if (foo()) {
        break A;  /*error The label 'A' is unnecessary.*/
    }
    bar();
}

B:
switch (a) {
    case 0:
        foo();
        break B;  /*error The label 'B' is unnecessary.*/
}
```

The following patterns are not considered problems:

```js
/*eslint no-extra-label: 2*/

A:
for (let i = 0; i < 10; ++i) {
    for (let j = 0; j < 10; ++j) {
        if (foo()) {
            break A;
        }
        bar();
    }
}

B:
switch (a) {
    case 0:
        for (let j = 0; j < 10; ++j) {
            if (foo()) {
                break B;
            }
        }
        break;
}

C: {
    if (foo()) {
        break C;
    }
    bar();
}
```

**Comment by nzakas** [comment]
Looks good

**Comment by mysticatea** [comment]
I'm working on this.
